src/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -324,4 +324,10 @@
   <data name="SerializerOptionsImmutable" xml:space="preserve">
     <value>Serializer options cannot be changed once serialization or deserialization has occurred.</value>
   </data>
+  <data name="SerializerPropertyNameConflict" xml:space="preserve">
+    <value>The property '{0}.{1}' has the same name as a previous property based on naming or casing policies.</value>
+  </data>
+  <data name="SerializerPropertyNameNull" xml:space="preserve">
+    <value>The property '{0}.{1}' name cannot be null as a result of naming policies.</value>
```

Comment:
> The property 'Foo.Bar' name cannot be null as a result of naming policies

This string is a little wonky IMO. I'd suggest something like:

> The property name for  'Foo.Bar' cannot be null as a result of naming policies

Reply:
On second thought, does it make sense to just use the current property name when a null is returned instead of an exception? This can occur when the converter returns null or when [JsonName] or [JsonName(null)] is used.

Reply:
Done